### PR TITLE
NIAD-2812: Send the COPC Multipart Content-Type header as "application/octet-stream"

### PR DESCRIPTION
**Pull request by adrianclay**
Within the "GP2GP Supplementary Specification"
  AKA "GP2GP: Handling Large Messages"
  AKA "NPFIT-PC-BLOD-0170.03 v1.1"
  AKA "HSCIC-FNT-TO-TAR-0095.02"

Under the section "AP2.5 Unsupported Content Types"

Content types which are unsupported by the Spine External Interface Specification should be shipped as "application/octet-stream" when sent as part of a COPC_IN010000UK01 message.

For the list of supported content-types within EIS, see
  Spine External Interface Specification Part 2
    Section 2.5.4.2. Attachments

https://digital.nhs.uk/developer/guides-and-documentation/spine-external-interface-specification

Annex 1:
Clinical Messaging – Attachment Types

### CODE REVIEW

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendDocumentTaskExecutor.java [hunk 1]
```diff
@@ -74,7 +74,7 @@ public void execute(SendDocumentTaskDefinition taskDefinition) {
                 var messageId = randomIdGeneratorService.createNewId();
                 var filename = mainMessageId + "_" + i + MESSAGE_ATTACHMENT_EXTENSION;
                 var chunkPayload = generateChunkPayload(taskDefinition, messageId, filename);
-                var chunkedOutboundMessage = createChunkOutboundMessage(chunkPayload, chunk, contentType);
+                var chunkedOutboundMessage = createChunkOutboundMessage(chunkPayload, chunk, "application/octet-stream");
```
Comment (adrianclay) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendDocumentTaskExecutor.java:77]: I'm suspicious that no tests failed when making this change. Will introduce a unit test which covers this behaviour.

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/SendDocumentTaskExecutorTest.java [hunk 2]
```diff
@@ -1,25 +1,145 @@
...
+                        .fromOdsCode("RANDOM-ODS")
+                        .conversationId("RANDOM-ID")
+                        .build()
+        );
+
+        verify(mhsRequestBuilder, times(NUMBER_OF_CHUNKS)).buildSendEhrExtractCommonRequest(
+                contains("\"content_type\":\"application/octet-stream\""),
+                eq("RANDOM-ID"),
+                eq("RANDOM-ODS"),
+                any()
```
Comment (ole4ryb) [inline service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/SendDocumentTaskExecutorTest.java:129]: I think slightly narrowing down the last param to anyString() is better than pass any()

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendDocumentTaskExecutor.java (hunk 1 above)
Comment (adrianclay) [reply service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendDocumentTaskExecutor.java:77]: This was resolved in #517

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendDocumentTaskExecutor.java [hunk 3]
```diff
@@ -73,7 +73,9 @@ public void execute(SendDocumentTaskDefinition taskDefinition) {
                 var messageId = randomIdGeneratorService.createNewId();
                 var filename = mainMessageId + "_" + i + MESSAGE_ATTACHMENT_EXTENSION;
                 var chunkPayload = generateChunkPayload(taskDefinition, messageId, filename);
-                var chunkedOutboundMessage = createChunkOutboundMessage(chunkPayload, chunk, taskDefinition.getDocumentContentType());
+
+                var chunkedOutboundMessage = createChunkOutboundMessage(chunkPayload, chunk, "application/octet-stream");
```
Comment (benhession) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendDocumentTaskExecutor.java:77]: Would it be better if the MIME type was a constant? As this is the pattern we've been using. See MESSAGE_ATTACHMENT_EXTENSION on line 74.

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/SendDocumentTaskExecutorTest.java (hunk 2 above)
Comment (ole4ryb) [reply service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/SendDocumentTaskExecutorTest.java:121]: So what about this point above?

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendDocumentTaskExecutor.java (hunk 3 above)
Comment (benhession) [reply service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendDocumentTaskExecutor.java:77]: Or the Tika static variable used in `DocumentToMHSTranslator.java`

**Review by benhession — APPROVED** [review]

**Review by ole4ryb — APPROVED** [review]

**Pull request merged**